server/src/routes/api/v1/ingredientsRouter.js

```diff
@@ -53,4 +53,88 @@ ingredientsRouter.post("/", async (req, res) => {
   }
 })
 
+ingredientsRouter.patch("/", async (req, res) => {
```

Comment:
There's a great opportunity here to create a helper class / model that contains methods for handling a lot of this logic to clean it up. This would help greatly with readability, and could mean that the endpoint ultimately has one `return res` statement rather than ~4 depending on how the patch is handled. Happy to chat about this if you're interested!